src/aspire/source/micrograph.py

```diff
@@ -0,0 +1,306 @@
+        Accesses and returns micrographs without any added noise.
+
+        :param indices: A 1-D NumPy array of integer indices.
+        :return: An `Image` object representing the clean micrograph
+        """
+        # Initialize empty micrograph
+        n_micrographs = len(indices)
+        clean_micrograph = np.zeros(
+            (n_micrographs, self.micrograph_size, self.micrograph_size),
+            dtype=self.dtype,
+        )
+        # Pad the micrograph
+        clean_micrograph = np.pad(
+            clean_micrograph,
+            ((0, 0), (self.pad, self.pad), (self.pad, self.pad)),
+            "constant",
+            constant_values=(0),
+        )
+        # Get centers
+        parity = self.particle_box_size % 2
+        for m in range(n_micrographs):
+            global_id = indices[m]
+            image = self.simulation.clean_images[self.get_particle(global_id)]
+            centers = self.centers[global_id]
+            x_lefts = centers[:, 0] - self.particle_box_size // 2 + self.pad
+            x_rights = centers[:, 0] + self.particle_box_size // 2 + parity + self.pad
+            y_lefts = centers[:, 1] - self.particle_box_size // 2 + self.pad
+            y_rights = centers[:, 1] + self.particle_box_size // 2 + parity + self.pad
+            # Subtract image from the micrograph using the particle's bounding box and center
+            for p in range(self.particles_per_micrograph):
+                clean_micrograph[m][
+                    x_lefts[p] : x_rights[p], y_lefts[p] : y_rights[p]
+                ] = (
+                    clean_micrograph[m][
+                        x_lefts[p] : x_rights[p], y_lefts[p] : y_rights[p]
+                    ]
+                    - image[p]
+                )
+            # Normalize micrographs
+            min_value = np.abs(np.min(clean_micrograph[m]))
+            clean_micrograph[m] = clean_micrograph[m] + min_value
+        clean_micrograph = clean_micrograph[
+            :,
+            self.pad : self.micrograph_size + self.pad,
+            self.pad : self.micrograph_size + self.pad,
+        ]
+        return Image(clean_micrograph)
+
+    def _images(self, indices):
+        """
+        Accesses and returns the projections from the Simulation.
+
+        :param indices: A 1-D NumPy array of integer indices.
+        :return: An `Image` object.
+        """
+        return self.simulation.images[indices]
+
+    def get_micrograph(self, particle_id):
+        """
+        Using the global ID of the particle, returns the micrograph ID and the local particle ID.
+
+        :param particle_id: Global ID of the particle.
+        :return: The micrograph ID and the local ID of the particle.
+        """
+        if particle_id >= self.total_particle_count or particle_id < 0:
+            raise RuntimeError("ID out of bounds.")
+        return divmod(particle_id, self.particles_per_micrograph)
+
+    def get_particle(self, micrograph_id, particle_id=None):
```

Comment:
Same here, maybe `get_particle_indices` (since we may have multiple in the default case).